In [1]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

In [2]:
df_ToolsAndImprovementRating_meta=pd.read_csv('ToolsAndHomeImprovement(meta).csv')

In [3]:
df_ToolsAndImprovementRating_meta.head()

category  \
0  ['Tools & Home Improvement', 'Lighting & Ceili...   
1  ['Tools & Home Improvement', 'Lighting & Ceili...   
2  ['Tools & Home Improvement', 'Paint, Wall Trea...   
3  ['Tools & Home Improvement', 'Lighting & Ceili...   
4                                                 []   

                                               tech1  \
0                                                NaN   
1   class="a-keyvalue prodDetTable" role="present...   
2                                                NaN   
3                                                NaN   
4   class="a-keyvalue prodDetTable" role="present...   

                                         description  fit  \
0                         ['collectible table lamp']  NaN   
1  ['Fun book light! Comes with two AAA batteries...  NaN   
2  ['A fun addition to any decor, The Beatles Yel...  NaN   
3  ['SMOKE FREE WAREHOUSE. Sized perfectly for re...  NaN   
4                                                 []  NaN   

                                               title  \
0                       Everett's Cottage Table Lamp   
1                    Diary of a Wimpy Kid Book Light   
2  Mudpuppy The Beatles Yellow Submarine Wall Decals   
3                       Lightwedge Paperback - Black   
4      Rob Cosman &quot;Hand-Cut Dovetails&quot; DVD   

                                            also_buy tech2           brand  \
0                                                 []   NaN             NaN   
1                                                 []   NaN  Barnes & Noble   
2  ['1481403621', 'B00EMLN7PS', 'B077NNCBTP', 'B0...   NaN        Mudpuppy   
3  ['1223027090', 'B007D3N6KS', '1452135215', '06...   NaN  LightWedge LLC   
4                                                 []   NaN      ROB COSMAN   

                                             feature  \
0                                                 []   
1  ['Easily clips to hardcover and paperback book...   
2  ['6 sheets of decals', 'Package: 11.25 x 11.25...   
3                                                 []   
4                                                 []   

                                                rank  \
0  ['>#3,780,135 in Tools & Home Improvement (See...   
1  ['>#1,074,903 in Tools & Home Improvement (See...   
2  ['>#105,697 in Toys & Games (See Top 100 in To...   
3                                 118,253 in Books (   
4  ['>#1,650,914 in Tools & Home Improvement (See...   

                                           also_view  \
0                                                 []   
1                                                 []   
2  ['B003VYAHRI', 'B00EMLN7PS', 'B01GWKH0FO', '07...   
3  ['1223027090', 'B00O8LRECY', '0641077491', 'B0...   
4                                     ['0980941237']   

                   main_cat  \
0  Tools & Home Improvement   
1  Tools & Home Improvement   
2              Toys & Games   
3                     Books   
4  Tools & Home Improvement   

                                        similar_item              date  \
0                                                NaN  October 30, 2010   
1                                                NaN     March 9, 2013   
2   class="a-bordered a-horizontal-stripes  a-spa...               NaN   
3                                                NaN               NaN   
4                                                NaN  February 8, 2010   

    price        asin                                           imageURL  \
0     NaN  001212835X                                                 []   
1     NaN  0594510384  ['https://images-na.ssl-images-amazon.com/imag...   
2  $20.52  0735342989  ['https://images-na.ssl-images-amazon.com/imag...   
3  $24.95  0972970169                                                 []   
4     NaN  0980941210                                                 []   

                                     imageURLHighRes details  
0                                 

In [4]:
df_ToolsAndImprovementRating_meta.shape

(571535, 19)

In [5]:
df_ToolsAndImprovementRating_meta.isnull().sum()

category                0
tech1              130672
description             0
fit                571088
title                   8
also_buy                0
tech2              571520
brand               16123
feature                 0
rank                    0
also_view               0
main_cat             1732
similar_item       342787
date               112869
price              238658
asin                    0
imageURL                0
imageURLHighRes         0
details                91
dtype: int64

In [6]:
df_ToolsAndImprovementRating_meta.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 571535 entries, 0 to 571534
Data columns (total 19 columns):
 #   Column           Non-Null Count   Dtype 
---  ------           --------------   ----- 
 0   category         571535 non-null  object
 1   tech1            440863 non-null  object
 2   description      571535 non-null  object
 3   fit              447 non-null     object
 4   title            571527 non-null  object
 5   also_buy         571535 non-null  object
 6   tech2            15 non-null      object
 7   brand            555412 non-null  object
 8   feature          571535 non-null  object
 9   rank             571535 non-null  object
 10  also_view        571535 non-null  object
 11  main_cat         569803 non-null  object
 12  similar_item     228748 non-null  object
 13  date             458666 non-null  object
 14  price            332877 non-null  object
 15  asin             571535 non-null  object
 16  imageURL         571535 non-null  object
 17  imageURLHi

In [7]:
df_ToolsAndImprovement_meta=df_ToolsAndImprovementRating_meta[['title','brand','main_cat','price','asin']]

In [8]:
df_ToolsAndImprovement_meta.head()

title           brand  \
0                       Everett's Cottage Table Lamp             NaN   
1                    Diary of a Wimpy Kid Book Light  Barnes & Noble   
2  Mudpuppy The Beatles Yellow Submarine Wall Decals        Mudpuppy   
3                       Lightwedge Paperback - Black  LightWedge LLC   
4      Rob Cosman &quot;Hand-Cut Dovetails&quot; DVD      ROB COSMAN   

                   main_cat   price        asin  
0  Tools & Home Improvement     NaN  001212835X  
1  Tools & Home Improvement     NaN  0594510384  
2              Toys & Games  $20.52  0735342989  
3                     Books  $24.95  0972970169  
4  Tools & Home Improvement     NaN  0980941210

In [9]:
df_ToolsAndImprovement_meta.isnull().sum()

title            8
brand        16123
main_cat      1732
price       238658
asin             0
dtype: int64

In [10]:
df_ToolsAndImprovement_meta.shape

(571535, 5)

In [11]:
df_ToolsAndImprovement_meta['main_cat'].value_counts()  #Sports & Outdoors,Office Products,Toys & Games,All Beauty,Health & Personal Care
#Home Audio & Theater,Musical Instruments,Pet Supplies,Camera & Photo,Cell Phones & Accessories,Car Electronics,Grocery,Books
#Video Games,Software,Portable Audio & Accessories                                                                                                                                                     26
#Movies & TV                                                                                                                                                                      22
#GPS & Navigation,Amazon Devices,Collectible Coins

Tools & Home Improvement                                                                                                                                                     433349
Amazon Home                                                                                                                                                                   58007
Industrial & Scientific                                                                                                                                                       32273
Sports & Outdoors                                                                                                                                                             14955
Automotive                                                                                                                                                                    12056
Office Products                                                                                     

#### Since we will be performing our analysis on patio and lawn items, we shall remove data belonginging to unnecessary categories like: Sports & Outdoors,Office Products,Toys & Games,All Beauty,Health & Personal Care,Home Audio & Theater,Musical Instruments,Pet Supplies,Camera & Photo,Cell Phones & Accessories,Car Electronics,Grocery,Books,Video Games,Software,Portable Audio & Accessories,Movies & TV,GPS & Navigation,Amazon Devices,Collectible Coins

In [12]:
lst=['Tools & Home Improvement','Amazon Home','Industrial & Scientific','Automotive','Arts, Crafts & Sewing','All Electronics','Computers','Appliances']

In [13]:
df2=df_ToolsAndImprovement_meta[df_ToolsAndImprovement_meta['main_cat'].isin(lst)]

In [14]:
df2

title  \
0                            Everett's Cottage Table Lamp   
1                         Diary of a Wimpy Kid Book Light   
4           Rob Cosman &quot;Hand-Cut Dovetails&quot; DVD   
5                   Rob Cosman Hand-Cut Dovetails 2.0 DVD   
6       Breeding Organic Vegetables: A Step-by-Step Gu...   
...                                                   ...   
571530  Small Vessel Sink White China Corner Above Cou...   
571531  Crqes Plastic Dual Flush Toilet Water Tank Pus...   
571532  Quietness @ Retro Lights Birdcage Iron Art Gla...   
571533  Solar Spotlight // MITHOS 50X Twin Warm White ...   
571534  Peakworks V8056022 PeakPro Positioning Belt-2D...   

                                 brand                  main_cat  \
0                                  NaN  Tools & Home Improvement   
1                       Barnes & Noble  Tools & Home Improvement   
4                           ROB COSMAN  Tools & Home Improvement   
5       RC Woodworking Hand Tools Inc.  Tools & Home Improvement   
6                             SioGreen  Tools & Home Improvement   
...                                ...                       ...   
571530              Renovator's Supply  Tools & Home Improvement   
571531                           Crqes  Tools & Home Improvement   
571532       Chandelier Lamp Quietness  Tools & Home Improvement   
571533                Solar Light Mart  Tools & Home Improvement   
571534                       Peakworks   Industrial & Scientific   

                price        asin  
0                 NaN  001212835X  
1                 NaN  0594510384  
4                 NaN  0980941210  
5                 NaN  0980966078  
6                 NaN  0982085028  
...               ...         ...  
571530        $115.00  B01HJG9LNY  
571531  $2.92 - $3.44  B01HJGM0O6  
571532            NaN  B01HJGVO16  
571533          $2.05  B01HJHS73S  
571534         $39.36  B01HJFRR40  

[540184 rows x 5 columns]

In [15]:
df2['main_cat'].value_counts()

Tools & Home Improvement    433349
Amazon Home                  58007
Industrial & Scientific      32273
Automotive                   12056
Arts, Crafts & Sewing         2843
All Electronics               1055
Computers                      535
Appliances                      66
Name: main_cat, dtype: int64

In [16]:
df2.isnull().sum()

title            6
brand        14458
main_cat         0
price       223173
asin             0
dtype: int64

In [17]:
for i in lst:
    print(i)
    df3=df2[df2['main_cat']==i]
    print(df3.shape)
    print(df3.isnull().sum())
    print('*'*50)

Tools & Home Improvement
(433349, 5)
title            5
brand         9895
main_cat         0
price       183816
asin             0
dtype: int64
**************************************************
Amazon Home
(58007, 5)
title           1
brand        4070
main_cat        0
price       25685
asin            0
dtype: int64
**************************************************
Industrial & Scientific
(32273, 5)
title          0
brand        264
main_cat       0
price       7694
asin           0
dtype: int64
**************************************************
Automotive
(12056, 5)
title          0
brand        130
main_cat       0
price       4009
asin           0
dtype: int64
**************************************************
Arts, Crafts & Sewing
(2843, 5)
title          0
brand         80
main_cat       0
price       1029
asin           0
dtype: int64
**************************************************
All Electronics
(1055, 5)
title         0
brand        13
main_cat      0
price       604
a

#### Categories like Computers andAll electronics have plenty of null values in their price column (>50%) which is an essential factor to take into consideration for analysis. Therefore we shall remove those categories aswell

In [18]:
lst2=['Computers','All Electronics']

In [19]:
df4=df2[~df2['main_cat'].isin(lst2)]

In [20]:
df4

title  \
0                            Everett's Cottage Table Lamp   
1                         Diary of a Wimpy Kid Book Light   
4           Rob Cosman &quot;Hand-Cut Dovetails&quot; DVD   
5                   Rob Cosman Hand-Cut Dovetails 2.0 DVD   
6       Breeding Organic Vegetables: A Step-by-Step Gu...   
...                                                   ...   
571530  Small Vessel Sink White China Corner Above Cou...   
571531  Crqes Plastic Dual Flush Toilet Water Tank Pus...   
571532  Quietness @ Retro Lights Birdcage Iron Art Gla...   
571533  Solar Spotlight // MITHOS 50X Twin Warm White ...   
571534  Peakworks V8056022 PeakPro Positioning Belt-2D...   

                                 brand                  main_cat  \
0                                  NaN  Tools & Home Improvement   
1                       Barnes & Noble  Tools & Home Improvement   
4                           ROB COSMAN  Tools & Home Improvement   
5       RC Woodworking Hand Tools Inc.  Tools & Home Improvement   
6                             SioGreen  Tools & Home Improvement   
...                                ...                       ...   
571530              Renovator's Supply  Tools & Home Improvement   
571531                           Crqes  Tools & Home Improvement   
571532       Chandelier Lamp Quietness  Tools & Home Improvement   
571533                Solar Light Mart  Tools & Home Improvement   
571534                       Peakworks   Industrial & Scientific   

                price        asin  
0                 NaN  001212835X  
1                 NaN  0594510384  
4                 NaN  0980941210  
5                 NaN  0980966078  
6                 NaN  0982085028  
...               ...         ...  
571530        $115.00  B01HJG9LNY  
571531  $2.92 - $3.44  B01HJGM0O6  
571532            NaN  B01HJGVO16  
571533          $2.05  B01HJHS73S  
571534         $39.36  B01HJFRR40  

[538594 rows x 5 columns]

In [21]:
lst2=[]
for i in df4['brand']:
    lst2.append(i)
set(lst2)

{nan,
 'WallWalker',
 'Winsor & Newton Regency Gold Series 520',
 'Wilmar',
 'Ustellar',
 'J. Hofert',
 'Hyjet Bidet',
 'Smart Meter',
 'Millennium Lighting',
 'Holabuyer',
 'Charming Customs',
 'imoocare',
 'Wennis Hardware Mfg. Ltd',
 'EasyEye',
 'BlueLED',
 "LAND O' LAKES, Inc. MN",
 'Commercial Electrtic',
 'Bestt Liebco',
 'Squirrel E-Mall',
 'Mesinton',
 'Wood Designs',
 'Amoker',
 'Streamlight',
 'Swylite',
 'Griffin',
 'Royalux',
 'HD Conservation - Watermiser',
 'Do it Best Faucets',
 'Perma-white',
 'Tru-Cut Galaxy',
 'A-link',
 'Adafruit',
 'Vatican Homes',
 'Awesome Products, Inc.',
 'Wood Owl',
 'Calimacil',
 'Graphic Products',
 'CONBRACO',
 'K-Four Switches',
 'Sanrio',
 'Fulldream',
 'Amertec-Westec',
 'superclosets.com',
 'Kult Kanvas',
 'Shrewd',
 'Tile By Design',
 'TwinCut',
 'Harper Home Products',
 'Eagle Access Control',
 'Patio Living Concepts',
 'Kisstaker',
 'Republic Diamond',
 'New Exakt Saw',
 'Golden Stag',
 'Jesada',
 'Albert Estate LTD',
 'CARESHINE',
 '

#### There doesn't seem to be any noise in the data so we can simply just remove null values

In [22]:
df4['brand'].isnull().sum()

14441

In [23]:
df5=df4.dropna(subset=['brand'])

In [24]:
df5['brand'].isnull().sum()

0

In [25]:
df5.shape

(524153, 5)

#### Cleaning basedcon title column

In [26]:
lst_title=[]
for i in df5['title']:
    lst_title.append(i)
set(lst_title)

{'RoomMates RMK1412GM Christmas Tree Quote Peel &amp; Stick Giant Wall Decal',
 nan,
 'Islamic Acrylic Car Hanging Ornament - Al Fatiha &amp; Ayat Al Kursi - Car Decoration - Hexagon - Green',
 'EZ-FLO 40073 Toilet Tank Flapper Economy',
 '3D Rose LSP_190805_1 USA, California. Reflections in a Mountain Lake. -Single Toggle Switch',
 'Hyjet Bidet',
 'FLIR IRW-4C Infrared Inspection Window, 4 in.',
 'Goof Off FG677 Super Glue Remover, 4-Ounce',
 'New Square D Thermal Overload Heater Element Unit B17.5 or B-17.5',
 'Believe in the vinyl wall lettering words sticky art home decor quotes stickers decals',
 '3/8&quot; X 100 Foot Goodyear Rubber Air Hose Compressor',
 'Carelove Remote Control Plug &amp; Outlet / Wireless Switch Kit for Household Appliances (3 set)',
 'ET2 E22322-61 Krisp 1-Light Wall Sconce, Polished Chrome Finish, White Glass, Mini Bi Pin T5 Fluorescent Fluorescent Bulb, 20W Max., Dry Safety Rated, 3000K Color Temp., Standard Dimmable, Glass Shade Material, 500 Rated Lumens,

#### There doesn't seem to be any noise present in our title so we can simply remove the null values

In [27]:
df5['title'].isnull().sum()

6

In [28]:
df6=df5.dropna(subset=['title'])

In [29]:
df6.isnull().sum()

title            0
brand            0
main_cat         0
price       209341
asin             0
dtype: int64

In [30]:
df6.dtypes

title       object
brand       object
main_cat    object
price       object
asin        object
dtype: object

In [31]:
df6['price']=df6['price'].fillna('0')

In [32]:
df6['price'].isnull().sum()

0

In [33]:
df6['price']=df6['price'].str.replace('$','')

In [34]:
df6['price']=df6['price'].str.replace(',','')

In [35]:
p=[]
for i in df6['price']:
    if len(i)<10:
        p.append(i)
    else:
        p.append('0')
p

['0',
 '0',
 '0',
 '0',
 '0',
 '31.90',
 '0',
 '129.99',
 '134.95',
 '129.99',
 '150.00',
 '0',
 '24.28',
 '1.96',
 '26.30',
 '0',
 '0',
 '0',
 '0',
 '0',
 '13.31',
 '0',
 '0',
 '0',
 '19.95',
 '14.39',
 '0',
 '7.56',
 '9.99',
 '14.99',
 '9.99',
 '0',
 '9.99',
 '0',
 '0',
 '0',
 '0',
 '9.98',
 '0',
 '16.89',
 '14.98',
 '18.99',
 '11.99',
 '20.90',
 '6.08',
 '0',
 '14.60',
 '10.99',
 '0',
 '12.29',
 '19.99',
 '5.99',
 '0',
 '0',
 '0',
 '0',
 '31.18',
 '8.00',
 '0',
 '70.95',
 '57.09',
 '16.96',
 '10.50',
 '4.88',
 '7.41',
 '0',
 '7.50',
 '0',
 '1.41',
 '10.95',
 '0',
 '0',
 '1.41',
 '16.95',
 '11.78',
 '6.99',
 '7.95',
 '10.95',
 '29.95',
 '7.95',
 '6.95',
 '0',
 '18.95',
 '10.95',
 '16.95',
 '0',
 '10.95',
 '16.99',
 '1.41',
 '0',
 '1.41',
 '0',
 '0',
 '10.99',
 '0',
 '19.68',
 '33.83',
 '7.86',
 '7.95',
 '16.95',
 '2.85',
 '0',
 '14.95',
 '9.99',
 '0',
 '0',
 '1.41',
 '0',
 '10.99',
 '16.26',
 '0',
 '0',
 '6.99',
 '0',
 '0',
 '12.95',
 '11.95',
 '9.95',
 '9.99',
 '1.41',
 '0',
 '0',
 

In [36]:
df6['price']=p

In [38]:
df6.dtypes

title       object
brand       object
main_cat    object
price       object
asin        object
dtype: object

In [39]:
df6['price']=df6['price'].astype('float64')

In [40]:
df6.dtypes

title        object
brand        object
main_cat     object
price       float64
asin         object
dtype: object

In [41]:
df6['price'].value_counts()

0.00       213721
1.41        13783
2.85         5125
9.99         3708
6.71         3555
            ...  
382.73          1
728.18          1
180.08          1
97.29           1
1990.00         1
Name: price, Length: 22045, dtype: int64

Tools & Home Improvement    433349
Amazon Home                  58007
Industrial & Scientific      32273
Automotive                   12056
Arts, Crafts & Sewing         2843
All Electronics               1055
Computers                      535
Appliances                      66

In [46]:
df6['main_cat'].value_counts()

Tools & Home Improvement    423449
Amazon Home                  53936
Industrial & Scientific      32009
Automotive                   11926
Arts, Crafts & Sewing         2763
Appliances                      64
Name: main_cat, dtype: int64

In [47]:
df_THI=df6[(df6['main_cat']=='Tools & Home Improvement')&(df6['price']!=0.00)]
print(df_THI['price'].mean())
df_AH=df6[(df6['main_cat']=='Amazon Home')&(df6['price']!=0.00)]
print(df_AH['price'].mean())
df_IS=df6[(df6['main_cat']=='Industrial & Scientific')&(df6['price']!=0.00)]
print(df_IS['price'].mean())
df_A=df6[(df6['main_cat']=='Automotive')&(df6['price']!=0.00)]
print(df_A['price'].mean())
df_ACS=df6[(df6['main_cat']=='Arts, Crafts & Sewing')&(df6['price']!=0.00)]
print(df_ACS['price'].mean())
df_AP=df6[(df6['main_cat']=='Appliances')&(df6['price']!=0.00)]
print(df_AP['price'].mean())

45.902273820588185
39.54119254461378
74.16798130222065
38.55676578514486
22.251686340640994
48.150000000000006


In [48]:
THI=df_THI['price'].mean()
AH=df_AH['price'].mean()
IS=df_IS['price'].mean()
A=df_A['price'].mean()
ACS=df_ACS['price'].mean()
AP=df_AP['price'].mean()

In [49]:
THI,AH,IS,A,ACS,AP

(45.902273820588185,
 39.54119254461378,
 74.16798130222065,
 38.55676578514486,
 22.251686340640994,
 48.150000000000006)

In [51]:
df_THI2=df6[df6['main_cat']=='Tools & Home Improvement']
df_AH2=df6[df6['main_cat']=='Amazon Home']
df_IS2=df6[df6['main_cat']=='Industrial & Scientific']
df_A2=df6[df6['main_cat']=='Automotive']
df_ACS2=df6[df6['main_cat']=='Arts, Crafts & Sewing']
df_AP2=df6[df6['main_cat']=='Appliances']

In [52]:
df_THI2['price'].replace(to_replace=0.00,value=THI,inplace=True)
df_AH2['price'].replace(to_replace=0.00,value=AH,inplace=True)
df_IS2['price'].replace(to_replace=0.00,value=IS,inplace=True)
df_A2['price'].replace(to_replace=0.00,value=A,inplace=True)
df_ACS2['price'].replace(to_replace=0.00,value=ACS,inplace=True)
df_AP2['price'].replace(to_replace=0.00,value=AP,inplace=True)

In [53]:
df_meta_concat_tools=pd.concat([df_THI2,df_AH2,df_IS2,df_A2,df_ACS2,df_AP2],axis=0)

In [54]:
df_meta_concat_tools

title  \
1                         Diary of a Wimpy Kid Book Light   
4           Rob Cosman &quot;Hand-Cut Dovetails&quot; DVD   
5                   Rob Cosman Hand-Cut Dovetails 2.0 DVD   
6       Breeding Organic Vegetables: A Step-by-Step Gu...   
8                                         IR Installation   
...                                                   ...   
521536  Chamberlain KLIK3U Clicker Universal Garage Do...   
521582  Multi-Code 4120 Gate/Garage Door Opener Remote...   
525446  Iso-Tip 7545 Fine Tip For Cordless Irons by Is...   
545009  Bostik 30812549 Self-Hardening Cementone Patio...   
564684  TurtleSkin Q3984LG Large CP Insider 330 Cut/Pu...   

                                 brand                  main_cat      price  \
1                       Barnes & Noble  Tools & Home Improvement  45.902274   
4                           ROB COSMAN  Tools & Home Improvement  45.902274   
5       RC Woodworking Hand Tools Inc.  Tools & Home Improvement  45.902274   
6                             SioGreen  Tools & Home Improvement  45.902274   
8                            Sio Green  Tools & Home Improvement  45.902274   
...                                ...                       ...        ...   
521536                     Chamberlain                Appliances  48.150000   
521582                      Multi-Code                Appliances  48.150000   
525446                         Iso-Tip                Appliances   6.770000   
545009                          Bostik                Appliances  21.650000   
564684                      TurtleSkin                Appliances  48.150000   

              asin  
1       0594510384  
4       0980941210  
5       0980966078  
6       0982085028  
8       0997438207  
...            ...  
521536  B018A36I2C  
521582  B018A39AQI  
525446  B0195UNAGG  
545009  B01CXA5BB2  
564684  B01GEAGR7Y  

[524147 rows x 5 columns]

In [55]:
df_meta_concat_tools['price'].value_counts()

45.902274      178566
39.541193       22281
1.410000        13783
74.167981        7835
2.850000         5125
                ...  
151.670000          1
668.990000          1
310.850000          1
1655.990000         1
392.140000          1
Name: price, Length: 22050, dtype: int64

In [56]:
header_info = ['product_id', 'Reviewer_id', 'rating', 'date_time']
df_ToolsAndImprovementRating=pd.read_csv('Tools_and_Home_Improvement (1).csv',names=header_info,header=None)

In [57]:
df_ToolsAndImprovementRating.head()

product_id     Reviewer_id  rating   date_time
0  0972970169  A11F6V4DD7WND4     3.0  1199664000
1  0972970169   AXPZYN4FP89L9     5.0  1173312000
2  0972970169   AOCT8ZWCRWLRE     4.0  1167782400
3  0972970169  A1UUD54HPAQZ2G     4.0  1150761600
4  0972970169  A2YHHBBJVTVD79     4.0  1105228800

In [58]:
df_meta_concat_tools.rename(columns={'asin':'product_id'},inplace=True)

In [59]:
df_meta_concat_tools.head()

title  \
1                    Diary of a Wimpy Kid Book Light   
4      Rob Cosman &quot;Hand-Cut Dovetails&quot; DVD   
5              Rob Cosman Hand-Cut Dovetails 2.0 DVD   
6  Breeding Organic Vegetables: A Step-by-Step Gu...   
8                                    IR Installation   

                            brand                  main_cat      price  \
1                  Barnes & Noble  Tools & Home Improvement  45.902274   
4                      ROB COSMAN  Tools & Home Improvement  45.902274   
5  RC Woodworking Hand Tools Inc.  Tools & Home Improvement  45.902274   
6                        SioGreen  Tools & Home Improvement  45.902274   
8                       Sio Green  Tools & Home Improvement  45.902274   

   product_id  
1  0594510384  
4  0980941210  
5  0980966078  
6  0982085028  
8  0997438207

In [60]:
tools=pd.merge(df_meta_concat_tools,df_ToolsAndImprovementRating,on='product_id',how='inner')

In [61]:
tools.isnull().sum()

title          0
brand          0
main_cat       0
price          0
product_id     0
Reviewer_id    0
rating         0
date_time      0
dtype: int64

In [62]:
tools.to_csv('tools_meta_ratingsonly_merged_and_cleaned.csv',index=False)